src/shared/varlink.c

```diff
@@ -1781,7 +1779,10 @@ static int varlink_format_json(Varlink *v, JsonVariant *m) {
         if (v->output_buffer_size + r + 1 > VARLINK_BUFFER_MAX)
                 return -ENOBUFS;
 
-        varlink_log(v, "Sending message: %s", text);
+        p = json_variant_by_key(m, "parameters");
+        sensitive = json_variant_is_sensitive(m) || json_variant_is_sensitive(ASSERT_PTR(p));
```

Comment:
hmm, it's not sufficient to check these two fields. we must go recursively down the tree.

maybe add a new flag JSON_FORMAT_REFUSE_SENSITIVE which makes json_variant_format() fail with a recognizable error if there's any sensitive field in the whole object tree. Then we'd call it twice: first with the flag. If it works, great we use it and know it's not sensitive. If it fails we call it again without the flag, and know it is sensitive.

By doing this via json_variant_format() we don#t need to scan the whole tree twice in the usual case, only if the thing is marked sensitive, which hopefully is the exception, not the common case.

Reply:
ok, implemented as requested